src/main/java/soot/FastHierarchy.java

```diff
@@ -53,6 +55,8 @@
  */
 public class FastHierarchy {
   private static final Logger LOGGER = LoggerFactory.getLogger(FastHierarchy.class);
+  private final LargeNumberedMap<Type, SmallNumberedMap<SootMethod>> typeToVtbl =
```

Comment:
Please change to Map<SootClass, Map<String,SootMethod>> and use HashMaps as implementation